### PR TITLE
Fixes #1570 #1572

### LINKED ISSUE #1570: retriever ls -v command does not work
While executing the ```retriever ls -v``` command, output is incorrect due to ```IndexError: list index out of range```.
Possible reasons are that some dataset json files have no license field or the license field is empty.
```
aakash01@aakash01-G3-3579:~/Desktop/gsoc/retriever (main)$ retriever ls -v
Offline datasets : 251

1. Abalone Age and Size Data
 abalone-age
[]
Database to aid in the prediction of the age of an Abalone given physical measurements
None
Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.

2. Long term limnological measures in Acton Lake, database, 1992-2017
 acton-lake
['lake', 'stream', 'nutrients', 'nitrogen', 'phosphorus', 'chlorophyll', 'LTREB', 'long-term', 'ecosystem', 'ammonium-nitrogen', 'nitrate-N', 'oxygen', 'carbon']
Long-term data collected from Acton Lake and its inflow streams, on a suite of physical, chemical and biological variables investigating how Acton Lake, responds to changes in ecosystem subsidies of detritus (sediments) and nutrients.
None
Michael J Vanni, Maria J Gonzalez, and William H Renwick. 2019. Long term limnological measures in Acton Lake, a southwest Ohio reservoir, and its inflow streams: 1992-2017. Environmental Data Initiative.

Traceback (most recent call last):
  File "/home/aakash01/anaconda3/envs/gsoc-dev/bin/retriever", line 8, in <module>
    sys.exit(main())
  File "/home/aakash01/anaconda3/envs/gsoc-dev/lib/python3.7/site-packages/retriever/__main__.py", line 150, in main
    licenses=str(script.licenses[0]['name']),
IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by Aakash3101**
Changes other than ```get_opts.py```

### CODE REVIEW

File: retriever/lib/datasets.py [hunk 1]
```diff
@@ -70,5 +70,36 @@ def license(dataset):
...
-    license_values = [str(script.licenses[0]['name']).lower() for script in SCRIPT_LIST()]
+    license_values = [
+        str(script.licenses[0]['name']).lower()
+        for script in SCRIPT_LIST()
+        if script.licenses
+    ]
     return set(license_values)
+
+
+def dataset_verbose_list(script_names: list):
```
Comment (henrykironde) [inline retriever/lib/datasets.py:81]: Where are we using this function?

Comment (Aakash3101) [reply retriever/lib/datasets.py:81]: It has the same implementation as ```retriever ls -v``` command as you requested.

Comment (henrykironde) [reply retriever/lib/datasets.py:81]: I don't see the function being called

Comment (Aakash3101) [reply retriever/lib/datasets.py:81]: It is like the ```datasets(keywords=None, licenses=None)``` function, like the ```datasets``` function returns datasets with specified keywords and licenses, the ```dataset_verbose_list``` prints the verbose of the specified dataset names

Comment (Aakash3101) [reply retriever/lib/datasets.py:81]: Any suggestions or changes @henrykironde?

Comment (henrykironde) [reply retriever/lib/datasets.py:81]: You need to use that new function in the main file

Comment (Aakash3101) [reply retriever/lib/datasets.py:81]: Got it. Thanks @henrykironde

**Comment by Aakash3101** [comment]
Used the ```dataset_verbose_list``` function to ```__main__.py```.
Here is an example:
``` python
(gsoc-dev) aakash01@aakash01-G3-3579:~/Desktop/gsoc/retriever (verbose-error)$ retriever ls -v airports
1. Dataset containing information on all airports on ouraiports.com
Dataset Name: airports
Keywords: ['airports', 'aeroplanes', 'airplanes', 'frequencies', 'runways', 'navaids', 'countries', 'regions']
Description: Dataset containing information on all airports on ourairports.com
Licenses: None
Citation: OurAirports.com, Megginson Technologies Ltd.

(gsoc-dev) aakash01@aakash01-G3-3579:~/Desktop/gsoc/retriever (verbose-error)$ retriever ls -v airports acton-lake
1. Dataset containing information on all airports on ouraiports.com
Dataset Name: airports
Keywords: ['airports', 'aeroplanes', 'airplanes', 'frequencies', 'runways', 'navaids', 'countries', 'regions']
Description: Dataset containing information on all airports on ourairports.com
Licenses: None
Citation: OurAirports.com, Megginson Technologies Ltd.

2. Long term limnological measures in Acton Lake, database, 1992-2017
Dataset Name: acton-lake
Keywords: ['lake', 'stream', 'nutrients', 'nitrogen', 'phosphorus', 'chlorophyll', 'LTREB', 'long-term', 'ecosystem', 'ammonium-nitrogen', 'nitrate-N', 'oxygen', 'carbon']
Description: Long-term data collected from Acton Lake and its inflow streams, on a suite of physical, chemical and biological variables investigating how Acton Lake, responds to changes in ecosystem subsidies of detritus (sediments) and nutrients.
Licenses: None
Citation: Michael J Vanni, Maria J Gonzalez, and William H Renwick. 2019. Long term limnological measures in Acton Lake, a southwest Ohio reservoir, and its inflow streams: 1992-2017. Environmental Data Initiative.

(gsoc-dev) aakash01@aakash01-G3-3579:~/Desktop/gsoc/retriever (verbose-error)$ retriever ls -v airport
Dataset(s) is not found
```

**Comment by henrykironde** [comment]
You need to also use the other interface `datasets(keywords=None, licenses=None)`

**Comment by Aakash3101** [comment]
> You need to also use the other interface `datasets(keywords=None, licenses=None)`

Can you please explain more, I am not able to understand where to use this interface, also it is already used in ```__main__.py``` at  [line](https://github.com/weecology/retriever/blob/main/retriever/__main__.py#L172)

**Comment by henrykironde** [comment]
Okey this looks good. I think every aspect is covered. I will go through it later, and let you know. Give me a day at least to review it.

**Comment by Aakash3101** [comment]
Sure

**Pull request closed without merging**